### PR TITLE
Clean up CreateFile usage.

### PR DESCRIPTION
**Pull request by JeremyKuhne**
The "safe" create file was specifically targetted at FileStream and doesn't apply to remaining usages in FileSystem.

Also collapse a few unnecessary methods.

### CODE REVIEW

File: src/Common/src/Interop/Windows/kernel32/Interop.CreateFile.cs [hunk 1]
```diff
@@ -15,38 +15,40 @@ internal partial class Kernel32
...
 
-        internal static SafeFileHandle CreateFileDefaultSecurity(
+        internal unsafe static SafeFileHandle CreateFile(
             string lpFileName,
             int dwDesiredAccess,
             FileShare dwShareMode,
             FileMode dwCreationDisposition,
             int dwFlagsAndAttributes)
         {
-            // This is technically equivalent to passing a null SECURITY_ATTRIBUTES
```
Comment (danmosemsft) [inline src/Common/src/Interop/Windows/kernel32/Interop.CreateFile.cs]: any idea why it didn't pass null before?

File: src/Common/src/Interop/Windows/kernel32/Interop.CreateFile.cs [hunk 2]
```diff
@@ -15,38 +15,40 @@ internal partial class Kernel32
...
             FileShare dwShareMode,
-            [In] ref SECURITY_ATTRIBUTES securityAttrs,
+            ref SECURITY_ATTRIBUTES securityAttrs,
             FileMode dwCreationDisposition,
             int dwFlagsAndAttributes,
             IntPtr hTemplateFile)
         {
             lpFileName = PathInternal.EnsureExtendedPrefixOverMaxPath(lpFileName);
-            return CreateFilePrivate(lpFileName, dwDesiredAccess, dwShareMode, ref securityAttrs, dwCreationDisposition, dwFlagsAndAttributes, hTemplateFile);
+            fixed (SECURITY_ATTRIBUTES* sa = &securityAttrs)
```
Comment (danmosemsft) [inline src/Common/src/Interop/Windows/kernel32/Interop.CreateFile.cs]: what is the advantage of this over ref?

File: src/Common/src/Interop/Windows/kernel32/Interop.CREATEFILE2_EXTENDED_PARAMETERS.cs [hunk 3]
```diff
@@ -8,13 +8,13 @@ internal partial class Interop
...
     {
-        internal struct CREATEFILE2_EXTENDED_PARAMETERS
+        internal unsafe struct CREATEFILE2_EXTENDED_PARAMETERS
         {
             internal uint dwSize;
             internal uint dwFileAttributes;
             internal uint dwFileFlags;
             internal uint dwSecurityQosFlags;
-            internal IntPtr lpSecurityAttributes;
+            internal Interop.Kernel32.SECURITY_ATTRIBUTES* lpSecurityAttributes;
```
Comment (danmosemsft) [inline src/Common/src/Interop/Windows/kernel32/Interop.CREATEFILE2_EXTENDED_PARAMETERS.cs]: `Interop.Kernel32.` is not necessary

Comment (danmosemsft) [reply src/Common/src/Interop/Windows/kernel32/Interop.CREATEFILE2_EXTENDED_PARAMETERS.cs]: same in other places

File: src/Common/src/Interop/Windows/kernel32/Interop.CreateFile.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/Common/src/Interop/Windows/kernel32/Interop.CreateFile.cs]: Can pass null, which is the normal case.

File: src/Common/src/Interop/Windows/kernel32/Interop.CreateFile.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/Common/src/Interop/Windows/kernel32/Interop.CreateFile.cs]: Aversion to pointers most likely.

File: src/Common/src/Interop/Windows/kernel32/Interop.CREATEFILE2_EXTENDED_PARAMETERS.cs (hunk 3 above)
Comment (JeremyKuhne) [reply src/Common/src/Interop/Windows/kernel32/Interop.CREATEFILE2_EXTENDED_PARAMETERS.cs]: True, fixing.

File: src/Common/src/Interop/Windows/kernel32/Interop.CreateFile2.cs [hunk 4]
```diff
@@ -11,11 +11,58 @@ internal partial class Interop
...
+        {
+            Interop.Kernel32.CREATEFILE2_EXTENDED_PARAMETERS parameters;
+            parameters.dwSize = (uint)Marshal.SizeOf<Interop.Kernel32.CREATEFILE2_EXTENDED_PARAMETERS>();
+
+            parameters.dwFileAttributes = (uint)dwFlagsAndAttributes & 0x0000FFFF;
+            parameters.dwSecurityQosFlags = (uint)dwFlagsAndAttributes & 0x000F0000;
+            parameters.dwFileFlags = (uint)dwFlagsAndAttributes & 0xFFF00000;
+
+            parameters.hTemplateFile = hTemplateFile;
+            parameters.lpSecurityAttributes = securityAttrs;
```
Comment (danmosemsft) [inline src/Common/src/Interop/Windows/kernel32/Interop.CreateFile2.cs]: Fixed not needed as it's copied into this stack based struct?

File: src/System.IO.FileSystem/src/System/IO/Win32FileSystem.cs [hunk 5]
```diff
@@ -389,16 +387,12 @@ private static SafeFileHandle OpenHandle(string fullPath, bool asDirectory)
                 throw new ArgumentException(SR.Arg_PathIsVolume, "path");
             }
 
-            Interop.Kernel32.SECURITY_ATTRIBUTES secAttrs = default(Interop.Kernel32.SECURITY_ATTRIBUTES);
-            SafeFileHandle handle = Interop.Kernel32.SafeCreateFile(
+            SafeFileHandle handle = Interop.Kernel32.CreateFile(
```
Comment (danmosemsft) [inline src/System.IO.FileSystem/src/System/IO/Win32FileSystem.cs]: SafeCreateFile was checking for FILE_TYPE_DISK. Can we assume that here?

Comment (danmosemsft) [reply src/System.IO.FileSystem/src/System/IO/Win32FileSystem.cs]: I don't think we can. Why don't we need this? Was it just for CAS?

File: src/Common/src/Interop/Windows/kernel32/Interop.CreateFile2.cs (hunk 4 above)
Comment (JeremyKuhne) [reply src/Common/src/Interop/Windows/kernel32/Interop.CreateFile2.cs]: Need to as it was passed in:

``` C#
        public unsafe static void Do(ref Guid guid)
        {
            // Complains
            DoNext(&guid);

            // This is ok
            Guid localGuid = new Guid();
            DoNext(&localGuid);
        }

        public unsafe static void DoNext(Guid* guid)
        {

        }
```

File: src/System.IO.FileSystem/src/System/IO/Win32FileSystem.cs (hunk 5 above)
Comment (JeremyKuhne) [reply src/System.IO.FileSystem/src/System/IO/Win32FileSystem.cs]: We don't need the check here- it was for creating FileStream, which is no longer in this repo. That said I don't intend to keep it for FileStream either. It goes back to trying to enforce limited trust- and we do anything similar for Unix.

**Pull request merged**